### ISSUE TITLE
automated-code-quality-gates: coverage threshold read only from the root jest config, which an Nx workspace keeps empty

### ISSUE THREAD

**Issue by ruleofcodedev** [Rule-Of-Code/RoC#98]
**Version:** 7.20.0
**Law:** `automated-code-quality-gates`

## What happens

```
Missing test coverage quality gates
```

`analyzeCoverageGates` looks for `jest.config.js|ts|json` **at the project root**
and reads the threshold from there:

```js
// laws/deployment/automated-code-quality-gates.js
const jestConfigFiles = ['jest.config.js', 'jest.config.ts', 'jest.config.json'];
for (const configFile of jestConfigFiles) {
  if (this.checkJestConfigFile(projectRoot, configFile)) { … break; }
}
```

In an Nx workspace the root config cannot hold a threshold. It is an aggregator,
and this is the whole file:

```ts
// jest.config.ts — repo root
export default async (): Promise<Config> => ({
  projects: await getJestProjectsAsync(),
});
```

The threshold lives with the project it applies to, which is where Jest expects
it in a multi-project setup:

```ts
// apps/web/jest.config.ts
coverageThreshold: {
  global: { branches: 100, functions: 100, lines: 100, statements: 100 },
},
```

That is the strictest threshold the format allows, and CI runs
`nx run web:test --coverage` against it on every push, so a regression below
100% fails the build. The law reports it as absent.

## Why it cannot be fixed on our side

Moving the threshold to the root config would not make it apply — Jest resolves
coverage thresholds per project when `projects` is used. Duplicating it there
would be a decoration that gates nothing, added only to be seen.

## Related

Same family as #94: a check that assumes a single-project layout and reads the
one place an Nx workspace deliberately keeps empty. The neighbouring sub-check
in this same law — code formatting gates — reads the CI file and found ours
immediately once we added `nx format:check`, so the law's other half already
works on this layout.

## Suggested direction

When the root Jest config resolves to `projects` (or `getJestProjectsAsync`),
follow it to the per-project configs and read the threshold there. `nx.json` next
to the root config is a reliable signal that this is the shape.

A cheaper first cut: also accept a `coverageThreshold` found in any
`**/jest.config.*` that the project's own build targets reference.

**Comment by ruleofcodedev** [comment]
**Fixed — in #99, awaiting CI and merge to `develop`. Ships in the next release.**

You were right that it could not be fixed on your side, and right about why: Jest resolves coverage thresholds per project when `projects` is used, so a threshold at the root would not apply. Duplicating it there **would be a decoration that gates nothing** — your phrase, and the reason that was never a candidate fix.

The law now asks the shared reader that looks across the workspace, so `apps/web/jest.config.ts` is found where Jest expects it. The root-only helper it used instead is deleted rather than left beside the fix.

The detail worth reporting back: **that shared reader already existed.** This law simply did not call it. Your "Related" note pointing at #94 was the useful part — it made us count, and this is the fifth law in two days reading a config from the root while an Nx workspace keeps it elsewhere, the third where the shared module was already written. The shape is not "a detector is wrong", it is "a detector did not ask the module that already knew", and that is a different thing to go hunting for.

Also noted: you told us the neighbouring sub-check already works on this layout. That is the kind of detail that stops a fix from being wider than it should be.

**Comment by ruleofcodedev** [comment]
Merged to `develop` in #99. Ships in the next release.
